### PR TITLE
Improved AMDJS merge implementation

### LINKED ISSUE #148: Resource filter for AMDJS throws NoSuchElementException
AMDJS merge implementation throws following exception.

```
java.util.NoSuchElementException: No line found
	at java.util.Scanner.nextLine(Scanner.java:1540)
	at com.ibm.g11n.pipeline.resfilter.impl.AmdJsResource.merge(AmdJsResource.java:276)
	at com.ibm.g11n.pipeline.tools.cli.ExportCmd._execute(ExportCmd.java:149)
	at com.ibm.g11n.pipeline.tools.cli.BaseCmd.execute(BaseCmd.java:103)
	at com.ibm.g11n.pipeline.tools.cli.GPCmd.main(GPCmd.java:80)
```

This problem is reproduced when the base JS file is ending with `});`. For example,
```
define({
    "key1": "value1",
    "key2": "value2"
});
```

### PR DESCRIPTION
**Pull request by yumaoka**
- Changed AMDJS merge implementation to use Rhino JS parser. Previous implementation used regex matcher for each line, and trying to replace value. This implementation was fragile and it does not work various different cases.
- Previous merge operation tried to wrap long lines, but the new implementation put translated value in a single line. We may revisit this and re-implement line wrapping logic with the new code base.
- Fixed #148 caused by the previous merge implementation.
- Previous AMDJS filter write/merge implementation did not escape string literal properly - such as `\n` for line feed, `\"` for quoataion and so on (note: parser worked OK because Rhino parser does unescaping). Added proper string literal escape code.
- Added new data drive test framework. With this framework, we can easily add parse/write/merge test cases without writing new JUnit test methods.
- Deleted old AMDJS test case files not used.

### CODE REVIEW

**Comment by srl295** [comment]
probably rebase on https://github.com/IBM-Cloud/gp-java-tools/pull/150 to fix the build issue

File: gp-res-filter/src/main/java/com/ibm/g11n/pipeline/resfilter/impl/AmdJsResource.java [hunk 1]
```diff
@@ -236,222 +293,132 @@ public void write(OutputStream outStream, LanguageBundle languageBundle,
...
+     * @return  JavaScript string literal expression.
      */
-    static String extractText(String text, int startIdx, int maxLen, BreakIterator brkItr) {
-        String s = text.substring(startIdx);
-        brkItr.setText(s);
-        int idx = brkItr.next();
-        while (true) {
-            int tmp = brkItr.next();
-            if (tmp == BreakIterator.DONE || tmp >= maxLen) {
+    static String escapeString(String str, Character quoteChar) {
```
Comment (srl295) [inline gp-res-filter/src/main/java/com/ibm/g11n/pipeline/resfilter/impl/AmdJsResource.java]: could this be done as part of gson?

Comment (yumaoka) [reply gp-res-filter/src/main/java/com/ibm/g11n/pipeline/resfilter/impl/AmdJsResource.java]: This is JS, not JSON. quote character might be quotation mark, or apostrophe.
There might be a standard library code to do this, but it's simple enough - so I just decided to implement it here.

Comment (srl295) [reply gp-res-filter/src/main/java/com/ibm/g11n/pipeline/resfilter/impl/AmdJsResource.java]: fair enough and good point.

**Pull request merged**